### ISSUE TITLE
How to install malabar-mode 1.x

### ISSUE THREAD

**Issue by erickj** [m0smith/malabar-mode#155]
Hi, thanks for the great work on malabar-mode, I've been in love with it for about a year.  Unfortunately I just upgraded to a new machine and am missing much of what I used most in 1.x in 2.x. After a few hours I've got a working 2.x install (I think), but am missing most of the old behavior.

Some of what I'm missing in 2.x:

C-h m - malabar mode is listed as a minor mode, and I don't see it's keybindings listed in the help menu.

C-c C-v C-c - compile file, I think this is working but I'm not sure. The groovy menu seems to output a very long list of compiled deps when it succeeds, and an error when there is an obvious syntax error, although I am missing the easy to read *Malabar compilation buffer* to get results. Also flycheck seems to compile the file continuously, not sure why

C-c C-v C-t - used to jump to the right unit test file for my project and run it (Junit4), now this no longer works. Even if I open the unit test directly, and issue C-c C-v t, I get an error "No runnable methods"

C-c C-v C-p - can't seem to jump to the POM anymore.

I'd love to help and add these to the 2.x branch but unfortunately I'm all thumbs with elisp and know nothing about groovy.

Given this, I'd like to downgrade to 1.x but see that it's been removed from melpa. What is the best way to install 1.x in light of that?

**Comment by erickj** [comment]
Just saw the 1.4.x branch install instructions, must have followed those a year ago, sorry for the noise.

**Comment by erickj** [comment]
Sorry again for the noise, I am unable to get the 1.x version installed from source. I have worked through a variety of errors (packagin, missing ~/.malabar_mode/classpath, etc...) but am now stuck on the issue described here:

https://github.com/m0smith/malabar-mode/issues/69

But don't see a final resolution in the ticket. Is there anyway to get an old 1.x version back in melpa to install from thanks?

Info about my enviornment:

$ mvn --version
Apache Maven 3.2.2 (NON-CANONICAL_2015-02-16T16:18:37_mockbuild; 2015-02-16T11:18:37-05:00)
Maven home: /usr/share/maven
Java version: 1.8.0_31, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.31-5.b13.fc21.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.18.8-201.fc21.x86_64", arch: "amd64", family: "unix"

$ java -version
openjdk version "1.8.0_31"
OpenJDK Runtime Environment (build 1.8.0_31-b13)
OpenJDK 64-Bit Server VM (build 25.31-b07, mixed mode)

$ emacs --version
GNU Emacs 24.4.1

**Comment by m0smith** [comment]
One of the primary reasons for rewriting was the old version only supported Maven 3.0.4.  Try using that version.

Also, thanks for the feedback on the missing features.  I have implemented the features that I use the most so it is good to know what other people are expecting.  I will create separate issues for them and get them fixed.

**Comment by erickj** [comment]
That's great (on creating separate issues). In the meanwhile I have a
sort-of working, sort-of-broken 1.x install. I'll just wait until the 2.x
branch is updated with the features I'm hoping for. I think running tests
directly is the biggest one, keeping me from switching.

On Wed, Mar 11, 2015 at 12:29 PM, Matthew O. Smith <notifications@github.com
> wrote:

> One of the primary reasons for rewriting was the old version only
> supported Maven 3.0.4. Try using that version.
>
> Also, thanks for the feedback on the missing features. I have implemented
> the features that I use the most so it is good to know what other people
> are expecting. I will create separate issues for them and get them fixed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/m0smith/malabar-mode/issues/155#issuecomment-78299654>
> .
>



-- 
-----
EJ

**Comment by erickj** [comment]
re: only supporting Maven 3.0.4 - that's odd because the setup on my previous machine had maven 3.2.2 installed (like my new machine). However I may have upgraded from maven 3.0.x at one point.

What I've noticed since forklift replacing the elpa package from my old laptop to my new laptop (which was the only way I found to get malabar to build the correct classpath file on start to work around #69), is that there seems to be some magical recipe of maven plugin JARs that the groovy shell is expecting in order to run the unit tests. For instance, it complained about me not having maven-resources-plugin version 2.5 (I had 2.6), when I downgraded to 2.5 in my POM, groovy then complained about not have surefire plugin 2.10 (I had 2.12). I tried going down this path for a while, just downgrading plugins that it seemed to want, but eventually hit a wall w/ a missing class I couldn't track down.

In any case, it's not worth tracking down any further, I'll just wait til the features I'm looking for hit 2.x.

Thanks again for the work on this project.
